protocol/messenger.go

```diff
@@ -5642,7 +5655,7 @@ func (m *Messenger) handleSyncClearHistory(state *ReceivedMessageState, message
 		return ErrChatNotFound
 	}
 
-	if existingChat.DeletedAtClockValue >= message.ClearedAt {
+	if existingChat.DeletedAtClockValue >= message.ClearedAt && (existingChat.DeletedAtClockValue != 0 && message.ClearedAt != 0) {
```

Comment:
This is horrible and shouldn't be needed.
I couldn't come up with a scenario however in which `DeleteAtClockValue` and `ClearedAt` were not `0`.

@cammellos @Samyoul maybe you can provide some insights here?

It's crucial we don't return here with `nil` because then the message state response won't have information about cleared histories..

Reply:
Turns out desktop was not using the correct RPC API which results in those values here being `0` when a sync clear message is received.